airbyte-webapp/src/components/connectorBuilder/YamlEditor/YamlEditor.tsx

```diff
@@ -54,7 +57,12 @@ export const YamlEditor: React.FC<YamlEditorProps> = ({ toggleYamlEditor }) => {
       try {
         const json = load(yamlValue) as ConnectorManifest;
         setYamlIsValid(true);
-        debouncedSetJsonManifest(json);
+        // skip setting the manifest on the first load as it just got passed in and is synced already
+        if (initialLoad.current) {
+          initialLoad.current = false;
```

Comment:
Should we have a similar thing for the builder form when switching from YAML -> UI?

It seems like currently it always calls `/update` whenever I switch from YAML to UI, even if the manifest is unchanged as a result of switching